core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractBox.java

```diff
@@ -249,4 +249,62 @@ public void setStorageLocation(StorageLocation storageLocation) {
     this.storageLocation = storageLocation;
   }
 
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + ((alias == null) ? 0 : alias.hashCode());
+    result = prime * result + (int) (boxId ^ (boxId >>> 32));
+    result = prime * result + ((creationTime == null) ? 0 : creationTime.hashCode());
+    result = prime * result + ((creator == null) ? 0 : creator.hashCode());
+    result = prime * result + ((description == null) ? 0 : description.hashCode());
+    result = prime * result + ((identificationBarcode == null) ? 0 : identificationBarcode.hashCode());
+    result = prime * result + ((locationBarcode == null) ? 0 : locationBarcode.hashCode());
+    result = prime * result + ((name == null) ? 0 : name.hashCode());
+    result = prime * result + ((size == null) ? 0 : size.hashCode());
+    result = prime * result + ((storageLocation == null) ? 0 : storageLocation.hashCode());
+    result = prime * result + ((use == null) ? 0 : use.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 44 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)